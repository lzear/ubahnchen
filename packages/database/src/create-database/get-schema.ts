import type BetterSqlite3 from 'better-sqlite3'

export const getSchema = (database: BetterSqlite3.Database) => {
  let schema = ''
  const rows = database
    .prepare(
      "SELECT type, name, sql FROM sqlite_master WHERE type IN ('table', 'index')",
    )
    .all() as { type: string; name: string; sql: string }[]

  for (const row of rows)
    schema += [`-- ${row.type.toUpperCase()} ${row.name}`, row.sql, ''].join(
      '\n',
    )
  return schema
}
