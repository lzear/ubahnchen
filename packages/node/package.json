{
  "name": "@ubahnchen/node",
  "version": "0.0.0",
  "type": "module",
  "exports": "./src/index.ts",
  "main": "src/index.ts",
  "scripts": {
    "build": "tsup",
    "lint": "eslint . --ext js,ts,jsx,tsx,cjs,json,md",
    "test": "node --loader ts-node/esm --es-module-specifier-resolution=node --experimental-vm-modules --no-warnings ../../node_modules/.bin/jest --config ../../jest.config.cjs --rootDir=./"
  },
  "dependencies": {
    "@ubahnchen/utils": "*",
    "ora": "^8.0.1",
    "pretty-bytes": "^6.1.1",
    "tmp": "^0.2.1"
  },
  "devDependencies": {
    "@jest/globals": "^29.7.0",
    "@types/tmp": "^0.2.6",
    "@ubahnchen/eslint-config": "*",
    "@ubahnchen/tsconfig": "*"
  },
  "volta": {
    "extends": "../../package.json"
  }
}