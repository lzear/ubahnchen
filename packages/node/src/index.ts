export { catFile, countLines } from './count-lines'
export {
  downloadFile,
  downloadOnce,
  downloadOnceOra,
  isAlreadyUpToDate,
} from './download-once'
export { fileSize, initializeFile, initializeFileIfNotExists } from './files'
export { Temporary } from './temporary'
