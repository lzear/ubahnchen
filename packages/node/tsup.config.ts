// eslint-disable-next-line import/no-extraneous-dependencies
import { defineConfig } from 'tsup'

import { defaultConfig } from '../../tsup.config.base.js'

export default defineConfig((options) => {
  return defaultConfig(
    {
      entry: ['./src/index.ts'],
    },
    options,
  )
})
