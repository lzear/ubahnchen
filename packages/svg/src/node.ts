export { findShortestPathStr } from './find-path/dijkstra-node'
export { annotateSvgFile } from './node/annotate'
export { filterSvgFile } from './node/filter'
export { mergePathsFile } from './node/merge-paths'
export { splitSubpaths } from './node/split-subpaths'
