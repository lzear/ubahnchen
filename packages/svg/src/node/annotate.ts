import { annotate } from '../annotate'

import { nodify } from './nodify'

export const annotateSvgFile = nodify(annotate)
