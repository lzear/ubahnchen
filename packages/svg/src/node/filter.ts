import { filterSvg } from '../filter'

import { nodify } from './nodify'

export const filterSvgFile = nodify(filterSvg)
