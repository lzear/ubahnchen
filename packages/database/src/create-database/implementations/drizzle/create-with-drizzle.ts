import * as path from 'node:path'
import * as url from 'node:url'

import BetterSqlite3 from 'better-sqlite3'
import { drizzle } from 'drizzle-orm/better-sqlite3'
import { type BetterSQLite3Database } from 'drizzle-orm/better-sqlite3'
import { migrate } from 'drizzle-orm/better-sqlite3/migrator'

import { initializeFile } from '@ubahnchen/node-utils'

type CreateWithSqlOptions = {
  filePath: string
}

export const createWithDrizzle = async ({
  filePath,
}: CreateWithSqlOptions): Promise<{
  database: BetterSqlite3.Database
  drizzleDatabase: BetterSQLite3Database
}> => {
  await initializeFile(filePath)
  const database = new BetterSqlite3(filePath)
  const drizzleDatabase = drizzle(database)
  const migrationsFolder = path.resolve(
    path.dirname(url.fileURLToPath(import.meta.url)),
    '../../../../migrations-folder',
  )
  migrate(drizzleDatabase, { migrationsFolder })
  return { database, drizzleDatabase }
}
