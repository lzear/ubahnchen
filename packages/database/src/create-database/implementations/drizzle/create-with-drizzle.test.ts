import { Temporary } from '@ubahnchen/node'

import { getSchema } from '../../get-schema'

import { createWithDrizzle } from './create-with-drizzle'

describe.skip(createWithDrizzle, () => {
  const temporary = new Temporary()
  it('creates with SQLite', async () => {
    const { database } = await createWithDrizzle({ filePath: temporary.file() })
    const schema = getSchema(database)
    expect(schema).toMatchSnapshot()
  })
})
