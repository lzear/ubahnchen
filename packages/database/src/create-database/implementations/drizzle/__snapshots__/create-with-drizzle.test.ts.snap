// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`createWithDrizzle > creates with SQLite 1`] = `
"-- TABLE __drizzle_migrations
CREATE TABLE "__drizzle_migrations" (
				id SERIAL PRIMARY KEY,
				hash text NOT NULL,
				created_at numeric
			)
-- INDEX sqlite_autoindex___drizzle_migrations_1

-- TABLE calendar
CREATE TABLE \`calendar\` (
	\`service_id\` text PRIMARY KEY NOT NULL,
	\`monday\` integer NOT NULL,
	\`tuesday\` integer NOT NULL,
	\`wednesday\` integer NOT NULL,
	\`thursday\` integer NOT NULL,
	\`friday\` integer NOT NULL,
	\`saturday\` integer NOT NULL,
	\`sunday\` integer NOT NULL,
	\`start_date\` text NOT NULL,
	\`end_date\` text NOT NULL
)
-- INDEX sqlite_autoindex_calendar_1

-- TABLE calendar_dates
CREATE TABLE \`calendar_dates\` (
	\`idx\` integer PRIMARY KEY NOT NULL,
	\`service_id\` text NOT NULL,
	\`date\` text NOT NULL,
	\`is_removing\` integer NOT NULL
)
-- TABLE routes
CREATE TABLE \`routes\` (
	\`route_id\` text PRIMARY KEY NOT NULL,
	\`route_name\` text NOT NULL,
	\`route_type\` integer NOT NULL
)
-- INDEX sqlite_autoindex_routes_1

-- TABLE shapes
CREATE TABLE \`shapes\` (
	\`shape_id\` text PRIMARY KEY NOT NULL,
	\`shape_pts\` text NOT NULL,
	\`lat_min\` real NOT NULL,
	\`lat_max\` real NOT NULL,
	\`lon_min\` real NOT NULL,
	\`lon_max\` real NOT NULL
)
-- INDEX sqlite_autoindex_shapes_1

-- TABLE stop_pairs
CREATE TABLE \`stop_pairs\` (
	\`idx\` integer PRIMARY KEY AUTOINCREMENT NOT NULL,
	\`stop_id_1\` text NOT NULL,
	\`stop_id_2\` text NOT NULL,
	\`route_id\` text NOT NULL,
	\`count\` integer NOT NULL,
	\`is_one_way\` integer NOT NULL,
	FOREIGN KEY (\`stop_id_1\`) REFERENCES \`stops\`(\`stop_id\`) ON UPDATE no action ON DELETE no action,
	FOREIGN KEY (\`stop_id_2\`) REFERENCES \`stops\`(\`stop_id\`) ON UPDATE no action ON DELETE no action,
	FOREIGN KEY (\`route_id\`) REFERENCES \`routes\`(\`route_id\`) ON UPDATE no action ON DELETE no action
)
-- TABLE sqlite_sequence
CREATE TABLE sqlite_sequence(name,seq)
-- TABLE stop_pairs_svg_paths
CREATE TABLE \`stop_pairs_svg_paths\` (
	\`stop_pair_idx\` integer NOT NULL,
	\`map\` text NOT NULL,
	\`waypoints\` text NOT NULL,
	\`length\` real NOT NULL,
	FOREIGN KEY (\`stop_pair_idx\`) REFERENCES \`stop_pairs\`(\`idx\`) ON UPDATE no action ON DELETE no action
)
-- TABLE stop_times
CREATE TABLE \`stop_times\` (
	\`idx\` integer PRIMARY KEY AUTOINCREMENT NOT NULL,
	\`trip_id\` text NOT NULL,
	\`arrival_time\` integer NOT NULL,
	\`departure_time\` integer NOT NULL,
	\`stop_id\` text NOT NULL,
	\`stop_sequence\` integer NOT NULL
)
-- TABLE stops
CREATE TABLE \`stops\` (
	\`stop_id\` text PRIMARY KEY NOT NULL,
	\`stop_name\` text NOT NULL,
	\`stop_lat\` real NOT NULL,
	\`stop_lon\` real NOT NULL,
	\`parent_station\` text,
	FOREIGN KEY (\`parent_station\`) REFERENCES \`stops\`(\`stop_id\`) ON UPDATE no action ON DELETE no action
)
-- INDEX sqlite_autoindex_stops_1

-- TABLE trips
CREATE TABLE \`trips\` (
	\`route_id\` text NOT NULL,
	\`shape_id\` text,
	\`service_id\` text NOT NULL,
	\`trip_id\` text PRIMARY KEY NOT NULL,
	\`trip_headsign\` text NOT NULL,
	\`start_time\` integer,
	\`end_time\` integer,
	FOREIGN KEY (\`route_id\`) REFERENCES \`routes\`(\`route_id\`) ON UPDATE no action ON DELETE no action
)
-- INDEX sqlite_autoindex_trips_1

-- INDEX route_name_idx
CREATE INDEX \`route_name_idx\` ON \`routes\` (\`route_name\`)
-- INDEX uniqueIdx
CREATE UNIQUE INDEX \`uniqueIdx\` ON \`stop_pairs\` (\`idx\`)
-- INDEX uniqueRouteStops
CREATE UNIQUE INDEX \`uniqueRouteStops\` ON \`stop_pairs\` (\`route_id\`,\`stop_id_1\`,\`stop_id_2\`)
-- INDEX uniquePerMap
CREATE UNIQUE INDEX \`uniquePerMap\` ON \`stop_pairs_svg_paths\` (\`stop_pair_idx\`,\`map\`)
"
`;
